### PR TITLE
Reduces Defib Safe Time: Adds Health HUD Defib Indicator

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Reduces the defib save time from 5 minutes down to 2 minutes.

Goal of this is to push defibs into the category of "correcting simple deaths that occur in medbay or on station" as opposed to being relied upon for general revival. Pushing more individuals to use cloning, day to day, will slow down the revival process (while not making it inherently more difficult) quite dramatically and therefore make death more "impactful".

This also adds in the ability to see when someone can still be defib'd. There will be an icon over their head that will persist until their defib timer expires.

Additionally, health analyzers will inform you how many seconds someone has been dead; at 120 seconds, they'll have to be cloned or revived with strange reagent

:cl: Fox McCloud
add: Health HUDs now have a defib indicator showing if they can be revived with a defib or not
tweak: Defib "safe" time reduced from 5 minutes to 2 minutes
fix: fixes a bug where sometimes revived individuals still appeared to be dead or severely damaged on health HUDs after being revived with a defib
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -162,6 +162,10 @@ REAGENT SCANNER
 	user.show_message("<span class='notice'>Body Temperature: [M.bodytemperature-T0C]&deg;C ([M.bodytemperature*1.8-459.67]&deg;F)</span>", 1)
 	if(M.timeofdeath && (M.stat == DEAD || (M.status_flags & FAKEDEATH)))
 		user.show_message("<span class='notice'>Time of Death: [station_time_timestamp("hh:mm:ss", M.timeofdeath)]</span>")
+		var/tdelta = round(world.time - M.timeofdeath)
+		if(tdelta < (DEFIB_TIME_LIMIT * 10))
+			user.show_message("<span class='danger'>Subject died [DisplayTimeText(tdelta)] ago, defibrillation may be possible!</span>")
```
Comment (Fox-McCloud) [inline code/game/objects/items/devices/scanners.dm]: One day, defibs need to not use this horrifying `show_message`.

**Comment by Spartan6** [comment]
While I’m all for making death more meaningful, this more punishes a few species harshly to the point that an early round death and even a slightly delayed retrieval can easily put you out the round permenantly.

I’d rather go down the route of cloning alternatives.

**Comment by datlo** [comment]
I'm not 100% sure on the reduced defib time because of unclonable species but definitely yes on the defib hud icon/health analyser change.

Altho if you choose to play an offbeat species, you gotta assume the increased possibility of malpracticebay...

**Comment by Dinarzad** [comment]
> While I’m all for making death more meaningful, this more punishes a few species harshly to the point that an early round death and even a slightly delayed retrieval can easily put you out the round permenantly.
> 
> I’d rather go down the route of cloning alternatives.

"Unable to be cloned" is an advertised negative to a species. It's meant to be a MAJOR downside, and it's one players should be well aware of. I fully accept that, as a slime person, if I die, odds are I am not coming back.
Shit happens when you work for NT, I'll live. (Or I guess I won't actually >_>) and I sure so will others.
There's always SR regardless.

**Comment by Raze500** [comment]
Nerfing the safe time? hell for that add another cloning bay...

**Comment by Shadeykins** [comment]
> 
> 
> While I’m all for making death more meaningful, this more punishes a few species harshly to the point that an early round death and even a slightly delayed retrieval can easily put you out the round permenantly.
> 
> I’d rather go down the route of cloning alternatives.

This is an honestly meaningful alternative to the other cloning PR. If we want death to actually have some measure of impact, clawing back the use of defibs as cure-alls (which they really can be right now) is a good first step.

"No-Clone" is intended to be a serious downside (especially when cloning was the only revival method), not something that can be handwaved.

**Comment by Citinited** [comment]
I'm going to echo what Shadeykins said - if you want death to be meaningful, adding more (viable) options to clone corpses goes completely against what you set out to do (it just makes death even more of a temporary sit out). Punishing noclone species for dying, on the other hand, *actually makes them consider if something they're about to do could kill them*.

You can't have your cake and eat it

**Comment by Fox-McCloud** [comment]
>Nerfing the safe time? hell for that add another cloning bay...

The whole reason this is being clawed back is precisely to increase dependency on the clone bay. While cloning doesn't necessarily require a lot of interaction it does use up a considerable amount of time, and as old-school players can attest, when the amount of deaths pile up, cloning can be painfully slow, especially when it can just get blown up.

Defibs and Strange reagent circumvent this paradigm; while they have a bit more involvement, they can still be utilized without a time sink and you can quickly revive a lot of people in short order; if we want death to have more "teeth" behind it, making the player wait much longer and adding risk to revival (oops, there goes our single point of failure) is a good way without necessarily punishing them if their bodies are found.

**Comment by EvadableMoxie** [comment]
I don't know how to feel about this when I don't know how other pending PRs are going to go.  Are we decreasing defib time to increase reliance on cloning, because we don't want to double clone time? Or are we doing both?  Are non-clonable species still going to have SR?  Or aren't they? Because the difference between a 2 minute and 5 minute defib time isn't nearly as huge a deal if SR exists than it is if it isn't.

These separate death PRs that all affect each other I feel are honestly the worst possible way to go about it. It's all interlinked, we should have one singular death overhaul PR instead of all these piecemeal PRs being considered in a vacuum but that might be implemented together. 

On that same topic, I don't like adding the defibbable icon to the med HUD onto this PR.  That's something non-controversial that pretty much everyone is going to want, but isn't really related to reducing the defib timer.  It feels like sugar coating something negative to try to make it more palatable when really it should be standing on it's own merits.

**Comment by Superhats** [comment]
I agree with this. People will argue that it cripples uncloneable species, but that's specifically the point of being uncloneable in the first place. It adds a major risk to dying, because you may not be revived at all.

**Pull request merged**